### PR TITLE
refactor: Use @wireapp/core to delete clients [FS-552]

### PR DESCRIPTION
**Pull request by atomrc**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-552" title="FS-552" target="_blank">TaskFS-552</a>  [Web] Register new client with MLS
  </td></table>
  <br />
 

With the integration of MLS we need to move all that is related to `clients` and `messages` to the core. 

This PR will let the core handle clients deletion (on backend and local db). In the future handling deletion of MLS clients will be completely transparent to the webapp

### CODE REVIEW

**Review by tlebon — APPROVED** [review]

**Review by thisisamir98 — APPROVED** [review]
LGTM

File: src/script/client/ClientRepository.ts [hunk 1]
```diff
@@ -287,10 +287,9 @@ export class ClientRepository {
    * @param password Password entered by user
    * @returns Resolves with the remaining user devices
    */
-  async deleteClient(clientId: string, password: string): Promise<ClientEntity[]> {
+  async deleteClient(clientId: string, password?: string): Promise<ClientEntity[]> {
     const selfUser = this.selfUser();
-    await this.clientService.deleteClient(clientId, password);
-    await this.deleteClientFromDb(selfUser.qualifiedId, clientId);
+    await this.core.service.client.deleteClient(clientId, password);
```
Comment (przemvs) [inline src/script/client/ClientRepository.ts:292]: This password can be undefined ?

Comment (atomrc) [reply src/script/client/ClientRepository.ts:292]: yes, that's on purpose. Temporary devices can be deleted without the password

Comment (przemvs) [reply src/script/client/ClientRepository.ts:292]: Okay, thanks!

**Review by przemvs — APPROVED** [review]

**Pull request merged**